docs/integrations/argocd-integration.md

```diff
@@ -0,0 +1,463 @@
+           command:
+           - /go/bin/terrascan
+           - version
+         periodSeconds: 10 
+       #if want to use private repo
+       volumeMounts:
+         - mountPath: /etc/github-secret
+           name: ssh-keys-github
+           readOnly: true
+         - mountPath: /etc/ssh-config-secret
+           name: ssh-config
+           readOnly: true
+         - mountPath: /etc/ssh-known-hosts-secret
+           name: ssh-known-hosts
+           readOnly: true
+         - mountPath: /data
+           name: notification-scripts 
+           readOnly: true
+ 
+     restartPolicy: Never
+ backoffLimit: 1
+``` 
+ 
+As shown, the PreSync requires access to the repository where IaC is stored, using the same branch (default) as the Argo CD application pipeline.
+Configuring the job to delete only after the specified time see ttlSecondsAfterFinished will allow users to check for violations in the User Interface, the alternative is through notifications.
+
+Example slack notification script
+
+```sh
+#!/bin/sh
+ 
+function send_slack_notificaton {
+  channel=$1
+  username=$2
+  slack_hook=$3
+ 
+  curl -X POST --data-urlencode payload="{\"channel\": \"#${channel}\", \"username\": \"${username}\", \"text\": \" \`\`\` $(cat results.out) \`\`\` \", \"icon_emoji\": \":ghost:\"}" ${slack_hook}
+}
+ 
+if [ -p /dev/stdin ]; then
+  echo "processing terrascan results"
+  while IFS= read line; do
+          echo "${line}" | tr '\\"' ' ' >> results.out
+  done
+ 
+  cat results.out
+ 
+  send_slack_notificaton $1 $2 $3
+ 
+  echo "notification exit code: $?"
+else
+  echo "no response skipping"
+fi
+``` 
+ 
+For non-public repositories, the private key, known_hosts and ssh config needs to be added as a kubernetes secret, configmap and secret respectively.
+ 
+```
+ kubectl create secret generic github-secret \
+   --from-file=ssh-privatekey= < path to your private key > \
+    --from-file=ssh-publickey=< path to your public key >
+```
+
+Config-map: 
+
+``` 
+  kubectl  create configmap ssh-known-hosts --from-file=< path to your known hosts file >
+```   
+
+``` 
+  kubectl  create configmap slack-notifications --from-file=< path to your notification script >
+```
+ 
+ssh config secret
+
+```
+ kubectl create secret generic ssh-config-secret \
+   --from-file=< path to your ssh config file >
+```   
+ 
+Example ssh config file
+
+``` 
+ Host github.com
+  HostName github.com
+  IdentityFile ~/.ssh/id_ed25519_github
+```
+
+Once the presynchook yaml file is completely configured, add this file to your repository folder for which Argo CD pipeline is configured.
+
+
+### 2. Use Terrascan Admission Webhook from PreSyncHook
+___
+You can use the already deployed terrascan’s k8s admission controller webhook to scan the remote repository from Argo CD PreSync hook.
+To configure, follow below steps
+
+
+#### Step 1: Configure terrascan admission controller webhook deployment yaml file with required keys and volumes and service to expose the controller pod.
```

Comment:
Rather than rendering these yamls , its better to point to a directory, guide or a blog ( Yusuf's blog for setting up the admission controller) . 

I have tried to create a crisp guide on setting up the terrascan server and webhook (if required) in my PR :
https://github.com/accurics/terrascan/pull/739/

With this:
1. We'll be able to follow DRY. 
2. We get a ready to use documentation with yaml files ready to be deployed to k8s.
3. Using kustomize helps in yaml manifests in a much neater way and doesn't really have a learning curve like helm.